File: src/codes/excessive-branching/py/uncode.py
def calculate_price(user_type: str, base_price: float) -> float:
    if user_type == "student":
        return base_price * 0.8
    elif user_type == "member":
        return base_price * 0.9
    elif user_type == "vip":
        return base_price * 0.7
    else:
        return base_price

# 利用例
print(calculate_price("student", 1000))  # 800

File: src/codes/excessive-branching/php/uncode.php
<?php
function calculatePrice(string $userType, float $basePrice): float {
  if ($userType === "student") {
    return $basePrice * 0.8;
  } elseif ($userType === "member") {
    return $basePrice * 0.9;
  } elseif ($userType === "vip") {
    return $basePrice * 0.7;
  } else {
    return $basePrice;
  }
}

// 利用例
echo calculatePrice("student", 1000) . PHP_EOL; // 800
